juju/juju
# Add suspend/resume relation support

## Description of change

Cross model relations to application offers can be suspended/resumed.
$ juju suspend-relation 123
$ juju resume-relation 123

A suspended relation will have the departed/broken hooks run but the relation remains in the model with the status "suspended". Similarly, a resumed relation will run the joined/changed hooks and the relation status becomes "joined".

## QA steps

Deploy a CMR scenario, mediawiki, mysql
On mysql model:
$ juju suspend-relation 1
verify that relation breaks, hooks are run using show-status-log, check list-offers shows suspended
$ juju resume-relation 1
verify that relation is repaired, hooks are run, list-offers shows joined
